src/components/enterprise-subsidy-requests/data/hooks.js

```diff
@@ -0,0 +1,83 @@
+import {
+  useState, useEffect, useMemo,
+} from 'react';
+import { logError } from '@edx/frontend-platform/logging';
+import { camelCaseObject } from '@edx/frontend-platform/utils';
+import { fetchSubsidyRequestConfiguration, fetchLicenseRequests, fetchCouponCodeRequests } from './service';
+import { SUBSIDY_TYPE } from '../constants';
+
+export function useSubsidyRequestConfiguration(enterpriseUUID) {
+  const [subsidyRequestConfiguration, setSubsidyRequestConfiguration] = useState();
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchCustomerConfiguration = async () => {
+      try {
+        const response = await fetchSubsidyRequestConfiguration(enterpriseUUID);
+        const config = camelCaseObject(response.data);
+        setSubsidyRequestConfiguration(config);
+      } catch (error) {
+        const httpErrorStatus = error.customAttributes?.httpErrorStatus;
+        if (httpErrorStatus === 404) {
+          // Customer configuration does not exist, subsidy requests are turned off.
+          setSubsidyRequestConfiguration(null);
+        } else {
+          logError(error);
+        }
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchCustomerConfiguration(enterpriseUUID);
+  }, [enterpriseUUID]);
+
+  return useMemo(() => ({ subsidyRequestConfiguration, isLoading }), [
+    subsidyRequestConfiguration,
+    isLoading,
+  ]);
+}
+
+export function useSubsidyRequests(subsidyRequestConfiguration) {
+  const [licenseRequests, setLicenseRequests] = useState([]);
+  const [couponCodeRequests, setCouponCodeRequests] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchSubsidyRequests = async (subsidyType) => {
+      setIsLoading(true);
+      try {
+        if (subsidyType === SUBSIDY_TYPE.COUPON) {
+          const { data: { results } } = await fetchCouponCodeRequests(
+            subsidyRequestConfiguration.enterpriseCustomerUuid,
+          );
+          const requests = camelCaseObject(results);
+          setCouponCodeRequests(requests);
+        } if (subsidyType === SUBSIDY_TYPE.LICENSE) {
+          const { data: { results } } = await fetchLicenseRequests(
+            subsidyRequestConfiguration.enterpriseCustomerUuid,
+          );
+          const requests = camelCaseObject(results);
+          setLicenseRequests(requests);
+        }
+      } catch (error) {
+        logError(error);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    if (subsidyRequestConfiguration?.subsidyRequestsEnabled) {
+      const { subsidyType } = subsidyRequestConfiguration;
+      if (subsidyType) {
+        fetchSubsidyRequests(subsidyType);
+      }
+    }
+  }, [subsidyRequestConfiguration?.subsidyRequestsEnabled]);
+
+  return useMemo(() => ({
+    couponCodeRequests,
+    licenseRequests,
+    isLoading,
+  }), [licenseRequests, couponCodeRequests, isLoading]);
```

Comment:
[clarification] Is this `useMemo` necessary versus returning `{ couponCodeRequests, licenseRequests, isLoading }` directly?